src/main/java/org/myrobotlab/service/interfaces/StateChangeHandler.java

```diff
@@ -0,0 +1,9 @@
+package org.myrobotlab.service.interfaces;
+
+import org.myrobotlab.service.FiniteStateMachine.StateChange;
+
+public interface StateChangeHandler {
+
+  public void handleStateChange(StateChange event);
```

Comment:
this should be onStateChange or something a little more consistent with the rest of the naming conventions in MRL.

Reply:
Oh I missed this one, yeah I agree. Could also change the name of the interface to StateChangeListener or something like that to match the other ones as well

Reply:
both good ideas updating